### PR TITLE
RFC: make julia prompt non bold

### PR DESCRIPTION
**Pull request by KristofferC**
Makes the julia prompt (`julia>`) non bold since the input and answer text now is non bold.

image

Lemme see some thumbs.

### CODE REVIEW

**Comment by StefanKarpinski** [comment]
I'd like some ability to turn thing bold again. E.g. `export JULIA_PROMPT_COLOR="bold_green"`. Can we do that currently? If not, could we add that feature to parsing of color names?

**Comment by KristofferC** [comment]
Yeah, that could be one reason to leave it on. Adding bold to all 16 + 256 colors is one way to do it (or punt on boldness for the 256 colors...).

**Comment by okvs** [comment]
Has there ever been any discussion about what the prompt actually should be? (Couldn't find any..)

Matlab has `>>` and Python has `>>>`. What about
screen shot 2017-01-26 at 15 16 40
?

(I apologize if this is just noise, but I believe this PR addresses the visual impact of the prompt, so I thought it relevant to bring it up. I always found the current prompt a bit 'heavyweight'.)

**Comment by StefanKarpinski** [comment]
What's wrong with `julia> ` as the prompt?

**Comment by okvs** [comment]
Nothing wrong, just a bit long :) If something more 'lightweight' would be considered, then `jl` is highly recognizable, given file extensions, package names, etc. Just throwing it out there, since visual impact was part of the consideration.

This can be changed by every user, of course -- though changing the length of the prompt seems to cause some cursor weirdness.

**Comment by ararslan** [comment]
Personally I like the extra visual distinction provided by having a bold prompt with non-bold input. Some terminal color schemes map colors in a weird way so in some cases the prompt color becomes nearly indistinguishable from the input color (Jameson noticed this with solarized dark), in which case having the bold by default is handy.

**Comment by KristofferC** [comment]
Isn't solarized mapping colours weirdly *because* of the bold?

**Comment by ararslan** [comment]
That I don't know, I don't use solarized. @vtjnash?

**Comment by pkofod** [comment]
(If matlab uses `>>`, python uses `>>>` the we should trump them and use `>>>>` and win bigly)

I vote for nonbold. Ohmyrepl has shown how costumizable the Julia repl is, so people can always turn bold back on (right?). Given that @KristofferC (I believe) has earlier show that color names are sometimes weird with bold on, I say make bold opt-in.

(Ps autocorrect wanted to change bigly to bigot, go figure)

**Comment by KristofferC** [comment]
The problem is that there is no really nice way of enabling bold + color once the prompt is non bold because the API so far only considered colors as changeable.

**Comment by Ismael-VC** [comment]
@pkofod *pypy* uses `>>>>`, should we go for `>>>>>` then? :smile:

 Seriously I like `jl>` as a prompt `julia>` is a little bit long for my taste. Currently one can change the prompt but the REPL doesn't behave correctly with prompts of different lenght than the original `julia>` as **okvs** mentions.

**Comment by StefanKarpinski** [comment]
I'm fine with making the prompt configurable, but I'm vehemently against changing the default from `julia>`. It's great marketing whenever someone demos something in Julia: what's that program that person is using to present this amazing demo? Well, it says "julia" right there.

**Comment by StefanKarpinski** [comment]
My proposal for an API for bold + color is to allow each color name to be prefixed with `bold_`.

**Comment by vtjnash** [comment]
> Isn't solarized mapping colours weirdly because of the bold?

Solarized maps all of the colors weirdly (it's designed to be low-contrast choice), so it's not much of a metric to go by for the average user.

**Comment by KristofferC** [comment]
BAM: https://kristofferc.github.io/OhMyREPL.jl/latest/features/prompt.html. Even got an output prompt.

**Comment by StefanKarpinski** [comment]
We should probably start merging some OhMyREPL features back into the Base REPL.

**Comment by KristofferC** [comment]
"start"?

https://github.com/JuliaLang/julia/pull/19680
https://github.com/JuliaLang/julia/pull/18228
https://github.com/JuliaLang/julia/pull/17599
https://github.com/JuliaLang/julia/pull/17838

;)

**Comment by StefanKarpinski** [comment]
Point taken. Let me rephrase that: we should continue to merge more awesome features from OhMyREPL into the main REPL 😁
